gateway/operations/postobject.go

```diff
@@ -114,9 +114,16 @@ func (controller *PostObject) HandleCompleteMultipartUpload(o *PathOperation) {
 		o.Log().WithError(err).Warn("could not delete multipart record")
 	}
 
-	// TODO: pass scheme instead of hard-coding http instead of https
+	// use url scheme and forwarded proto to identify http/https scheme
+	scheme := "http"
+	if o.Request.URL.Scheme == "https" {
+		scheme = "https"
+	} else if o.Request.Header.Get("X-Forwarded-Proto") == "https" {
```

Comment:
see - https://code.djangoproject.com/ticket/14597
There's another header for this we should add (also, maybe move this to a func in httputil?)